quickwit-cli/src/lib.rs

```diff
@@ -393,6 +407,39 @@ fn display_statistics(
     Ok(())
 }
 
+/// ThroughputCalculator is used to calculate throughput.
+struct ThroughputCalculator {
+    /// Stores the time series of processed bytes value.
+    processed_bytes_values: VecDeque<(Instant, usize)>,
+}
+
+impl ThroughputCalculator {
+    /// Creates new instance.
+    pub fn new(now: Instant) -> Self {
+        let processed_bytes_values: VecDeque<(Instant, usize)> =
+            (0..THROUGHPUT_WINDOW_SIZE).map(|_| (now, 0usize)).collect();
+        Self {
+            processed_bytes_values,
+        }
+    }
+
+    /// Calculates the throughput.
+    pub fn calculate(&mut self, current_processed_bytes: usize) -> f64 {
+        self.processed_bytes_values.pop_front();
+        let current_instant = Instant::now();
+        let (first_instant, first_processed_bytes) = *self.processed_bytes_values.front().unwrap();
+        let elapsed_time = (current_instant - first_instant).as_secs();
```

Comment:
The result here is equivalent to rounding to the closest second.
The time interval thing is not necessarily super accurate as it depends on the OS and tokio's scheduler, so it is better to work with a better precision.
`as_millis()` + cast as float and and divide the result by 1_000 should do the trick.